src/axom/quest/interface/internal/QuestHelpers.cpp

```diff
@@ -362,124 +368,228 @@ int read_c2c_mesh_uniform(const std::string& file,
+    reader.getMesh(static_cast<TetMesh*>(m));
   }
   else
   {
-    SLIC_WARNING("reading C2C file failed, setting mesh to NULL");
+    SLIC_WARNING("reading Pro/E file failed, setting mesh to NULL");
     delete m;
     m = nullptr;
-    revolvedVolume = 0.;
   }
 
   return rc;
 }
-#endif  // AXOM_USE_C2C
 
+#if defined(AXOM_USE_MFEM)
 /*
- * Reads in the Pro/E tetrahedral from the specified file.
+ * Reads in the contour mesh from the specified file.
  */
-int read_pro_e_mesh(const std::string& file, mint::Mesh*& m, MPI_Comm comm)
+int read_mfem_mesh(const std::string& file,
+                   bool uniform,
+                   const numerics::Matrix<double>& transform,
+                   int segmentsPerPiece,
+                   double vertexWeldThreshold,
+                   double percentError,
+                   mint::Mesh*& m,
+                   double& revolvedVolume)
 {
-  constexpr int DIMENSION = 3;
-  using TetMesh = mint::UnstructuredMesh<mint::SINGLE_SHAPE>;
+  // NOTE: MFEM meshes we are dealing with are always 2D
+  constexpr int DIMENSION = 2;
+  using SegmentMesh = mint::UnstructuredMesh<mint::SINGLE_SHAPE>;
 
-  // STEP 0: check input mesh pointer
+  // STEP 1: check input mesh pointer
+  revolvedVolume = 0.;
   if(m != nullptr)
   {
     SLIC_WARNING("supplied mesh pointer is not null!");
     return READ_FAILED;
   }
 
-  // STEP 1: allocate output mesh object
-  m = new TetMesh(DIMENSION, mint::TET);
+  // STEP 2: construct MFEM reader
+  quest::MFEMReader reader;
 
-  // STEP 2: construct Pro/E reader
-#ifdef AXOM_USE_MPI
-  quest::PProEReader reader(comm);
-#else
-  AXOM_UNUSED_VAR(comm);
-  quest::ProEReader reader;
-#endif
-
-  // STEP 3: read the mesh from the Pro/E file
+  // STEP 3: read the curves from the input file
+  axom::Array<axom::primal::NURBSCurve<double, 2>> curves;
   reader.setFileName(file);
-  int rc = reader.read();
+  int rc = reader.read(curves);
   if(rc == READ_SUCCESS)
   {
-    reader.getMesh(static_cast<TetMesh*>(m));
+    m = new SegmentMesh(DIMENSION, mint::SEGMENT);
+
+    // STEP 4: Make the linear segments.
+    LinearizeCurves lin;
+    lin.setVertexWeldingThreshold(vertexWeldThreshold);
+    if(uniform)
+    {
+      lin.getLinearMeshUniform(curves.view(), static_cast<SegmentMesh*>(m), segmentsPerPiece);
+    }
+    else
+    {
+      lin.getLinearMeshNonUniform(curves.view(), static_cast<SegmentMesh*>(m), percentError);
+    }
+    revolvedVolume = lin.getRevolvedVolume(curves.view(), transform);
   }
   else
   {
-    SLIC_WARNING("reading Pro/E file failed, setting mesh to NULL");
-    delete m;
+    SLIC_WARNING("reading MFEM file failed, setting mesh to NULL");
     m = nullptr;
   }
 
   return rc;
 }
+#endif
 
 /// Mesh Helper Methods
 
+#if defined(AXOM_USE_MFEM)
+template <typename CurveType, typename CreateFunc1, typename CreateFunc2>
+CurveType segment_to_curve_impl(const mfem::Mesh* mesh, int elem_id, CreateFunc1 &&create1, CreateFunc2 &&create2)
```

Comment:
Suggestion: 
`PolynomialConstructor` and `RationalConstructor` (or something similar) might be clearer than `CreateFunc1` and `CreateFunc2`.

Reply:
I changed the template args to the names you suggested.